### PR TITLE
Gidk 23ghp7 sales amount prod

### PR DESCRIPTION
**Pull request by shalinee-walkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

**Comment by shalinee-walkover** [comment]
https://app.clickup.com/t/23ghp7 - Make a separate component for the amount field and make same html in overall giddh

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.scss [hunk 1]
```diff
@@ -326,10 +326,13 @@ a[type="button"] {
             text-align: right;
             padding-bottom: 7px;
             display: table-cell;
+
         }
         .figure-cell {
             width: 140px;
             padding-right: 0;
+            // display: flex;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.scss:334]: Do remove unused code

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 2]
```diff
@@ -232,6 +233,8 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
     @ViewChild('itemsContainer', { read: ViewContainerRef, static: false }) container: ViewContainerRef;
     /** Template reference for each entry */
     @ViewChild('entry', { read: TemplateRef, static: false }) template: TemplateRef<any>;
+    @ViewChild('AmountFieldComponent', { static: true }) public amountField: ElementRef;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:236]: Please check if it is used as there is no reference in the TS file. If it is used then add JS doc

File: apps/web-giddh/src/app/services/general.service.ts [hunk 3]
```diff
@@ -777,4 +777,14 @@ export class GeneralService {
     public getFileExtension(path: string): string {
         return (path && path.match(/(?:.+..+[^\/]+$)/ig) != null) ? path.split('.').pop() : 'null';
     }
+    /** this will store currency code */
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/services/general.service.ts:780]: Do use Document this for adding JS Doc for methods

File: apps/web-giddh/src/app/shared/amount-field/amount-field.component.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import { GeneralService } from '../../services/general.service';
+
+@Component({
+    selector: 'amount-field',
+    templateUrl: './amount-field.component.html',
+    styleUrls: ['./amount-field.component.scss']
+})
+
+export class AmountFieldComponent implements OnInit, OnChanges {
+    @Input() public amount: any;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/amount-field/amount-field.component.ts:11]: Do add JS doc

File: apps/web-giddh/src/app/shared/amount-field/amount-field.component.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    @Input() public ratePrecision: boolean = true;
+
+    /** this will store direction */
+    public direction: string = "ltr";
+
+    constructor(private generalService: GeneralService) {
+
+    }
+
+    public ngOnInit(): void {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/amount-field/amount-field.component.ts:23]: Do add JS doc on all the methods with Document this

**Review by prateek-giddh — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.scss (hunk 1 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.scss:334]: Removed

File: apps/web-giddh/src/app/services/general.service.ts (hunk 3 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/services/general.service.ts:780]: Done

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts (hunk 2 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:236]: removed

File: apps/web-giddh/src/app/shared/amount-field/amount-field.component.ts (hunk 5 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/shared/amount-field/amount-field.component.ts:32]: Done

File: apps/web-giddh/src/app/shared/amount-field/amount-field.component.ts (hunk 4 above)
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/shared/amount-field/amount-field.component.ts:12]: Done

**Review by prateek-giddh — APPROVED** [review]

**Pull request merged**